### ISSUE TITLE
liquid_tags plugin does not support {filename} syntax for linking to internal content.

### ISSUE THREAD

**Issue by adworacz** [getpelican/pelican-plugins#178]
Per http://blog.getpelican.com/pelican-3.3-released.html, the {filename} or |filename| keyword prepended to a file path should resolve to the internal content.

More details about linking to internal content can be found here : http://docs.getpelican.com/en/3.3.0/getting_started.html#ref-linking-to-internal-content

That being said, the following tag is not compatible with the regex used in the **liquid_tags.video** plugin. Other liquid_tags plugins may be effected, but I have not tested them.

```
{% video {filename}/images/videos/video_hq.mp4 {filename}/images/videos/video_lq.mp4 640 480 %}
```

It catches the first link but fails to catch the second video or the width and height.

In short, the regex is not compatible with the {filename} notation and fails to resolve the {filename} PATH accordingly.
#### For reference:
- The _better_images_ plugin added support for it here: https://github.com/getpelican/pelican-plugins/commit/37809a83a8905f2cb0d3d54a50ebb5a69e36cae1
- And it is important to remember the **INTRASITE_LINK_REGEX** setting :
  http://docs.getpelican.com/en/3.3.0/settings.html

**Comment by kernc** [comment]
> That being said, I don't know how to look up the pelican instance's PATH settings from within your plugin.

@adworacz: Something like this would work:

```python

def other_handler(*args):
    # Use the PATH which was defined beforehand
    print PATH

def on_init(generator):
    global PATH
    PATH = generator.settings['PATH']

def register():
    signals.initialized.connect(on_init)
    signals.other_signal.connect(other_handler)
```

**Comment by justinmayer** [comment]
If someone would like to work on this, please feel free to comment here so the issue can be re-opened.
